profiles/serializers.py

```diff
@@ -259,6 +260,13 @@ class ProgramCertificateSerializer(serializers.ModelSerializer):
     Serializer for Program Certificates
     """
 
+    program_letter_url = serializers.SerializerMethodField()
+
+    def get_program_letter_url(self, instance):
+        return reverse(
+            "profile:program-letter-intercept", args=[instance.micromasters_program_id]
+        )
+
```

Comment:
This is returning urls in the format `"/program_letter/1/"`, which will only work for the certificate's logged-in user.  Wondering if maybe it should provide the shareable url instead if it exists yet (`/program_letter/<uuid>/view`), or both as separate fields (something like `create_letter_url`, `share_letter_url`) since it's possible the shareable url hasn't yet been created?  

Also, what do you think of returning absolute urls instead (`http://<domain>/program_letter/....`)?  The frontend could create a full url itself, but maybe easier to do it in the backend?

Reply:
@mbertrand - made it generate absolute urls and changed the url params to return both a program_letter_generate_url and a program_letter_share_url. I also made it so the serializer itself does a get_or_create for the program letter. In the future we might consider removing the intercept view altogether but I think it still has a place for now (i'm imagning a case where we send links to view view a program letter from some marketing automation software that is only aware of program ids etc)